src/pretix/base/services/orders.py

```diff
@@ -824,7 +825,8 @@ def _check_paid_price_change(self):
             raise OrderError(self.error_messages['paid_price_change'])
 
     def _check_paid_to_free(self):
-        if self.order.total == 0:
+        if self.order.total == 0 and (self._totaldiff < 0 or self.split_order and self.split_order.total > 0):
```

Comment:
1. Do you know the precedence rules of ``and`` and ``or`` without looking them up? I don't, so please add brackets here to make clear what you mean ;)
2. I just had to think about the logic of that line for at least 10 minutes. I got it now and I think it is correct, but would you mind adding an explanatory comment for the next one reading it? 😉